src/openrct2-ui/drawing/engines/opengl/TextureCache.cpp

```diff
@@ -61,6 +63,8 @@ void TextureCache::InvalidateImage(uint32_t image)
 
 BasicTextureInfo TextureCache::GetOrLoadImageTexture(uint32_t image)
 {
+    std::lock_guard<std::mutex> lock(_mutex);
```

Comment:
You only need the mutex if you are modifying the atlas right?

Reply:
It can't read while its being modified.

Reply:
wouldn't `shared_mutex` be more useful in such case?

Reply:
It would make more sense if the function didn't do both so it would over complicate things.

Reply:
Updated, even gave a nice increase of another 50 to 100 FPS ;)